lib/internal/module.js

```diff
@@ -3,11 +3,34 @@
 exports = module.exports = {
   makeRequireFunction,
   stripBOM,
-  addBuiltinLibsToObject
+  addBuiltinLibsToObject,
+  applyTransforms
 };
 
 exports.requireDepth = 0;
 
+const transforms = [];
+
+function addTransform(transform) {
+  transforms.push(transform)
+}
+function removeTransform(transform) {
+  const index = transforms.indexOf(transform)
+  if (index >= 0) {
+    transforms.splice(index, 1)
+  }
+}
+
+function applyTransforms(content, filename) {
+  return transforms.reduce((content, transform) => {
+    const next = transform(content, filename);
+    if (typeof next !== 'string') {
+      throw new Error('Module transforms must return the modified content')
+    }
+    return next;
+  }, content);
+}
+
```

Comment:
Event emitters are much bigger and more complicated than this needs.

That event emitter implementation has a lot more closures and object allocations. Event emitters also inherently have a much more complicated lookup scheme.

As require is a major startup hot-path, it's best not to put event emitters in the middle of that. Keep in mind this code would typically run hundreds to thousands of times during app startup.